Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -0,0 +1,687 @@
+                                                       Common::TCPHeader* tcpdata, u8* data)
+{
+  sf::IpAddress target;
+
+  if (tcpdata->properties & (TCP_FLAG_FIN | TCP_FLAG_RST))
+  {
+    int i = GetTCPSlot(tcpdata->source_port, tcpdata->destination_port,
+                       *(u32*)&ipdata->destination_addr);
+    if (i == -1)
+      return;  // not found
+
+    network_ref[i].ack_num++;
+    int size = BuildFINFrame(&network_ref[i], &m_in_frame[0]);
+    WriteToQueue(&m_in_frame[0], size);
+    network_ref[i].ip = 0;
+    network_ref[i].tcp_socket.disconnect();
+  }
+  else if (tcpdata->properties & TCP_FLAG_SIN)
+  {
+    // new connection
+    int i = GetTCPSlot(tcpdata->source_port, tcpdata->destination_port,
+                       *(u32*)&ipdata->destination_addr);
+    if (i != -1)
+      return;
+    i = GetAvaibleSlot(0);
+
+    network_ref[i].delay = 0x3000;
+    network_ref[i].local = tcpdata->source_port;
+    network_ref[i].remote = tcpdata->destination_port;
+    network_ref[i].ack_num = Common::swap32(tcpdata->sequence_number) + 1;
+    network_ref[i].ack_base = network_ref[i].ack_num;
+    network_ref[i].seq_num = 0x1000000;
+    network_ref[i].window_size = Common::swap16(tcpdata->window_size);
+    network_ref[i].type = IPPROTO_TCP;
+    for (auto& tcp_buf : network_ref[i].tcp_buffers)
+      tcp_buf.used = false;
+    network_ref[i].from.sin_addr.s_addr = *(u32*)&ipdata->destination_addr;
+    network_ref[i].from.sin_port = tcpdata->destination_port;
+    network_ref[i].to.sin_addr.s_addr = *(u32*)&ipdata->source_addr;
+    network_ref[i].to.sin_port = tcpdata->source_port;
+    memcpy(&network_ref[i].bba_mac, &m_eth_ref->mBbaMem[BBA_NAFR_PAR0], Common::MAC_ADDRESS_SIZE);
+    network_ref[i].my_mac = fake_mac;
+    network_ref[i].tcp_socket.setBlocking(false);
+
+    // reply with a sin_ack
+    memset(&m_in_frame[0], 0, 0x100);
+    Common::EthernetHeader* hwpart = (Common::EthernetHeader*)&m_in_frame[0];
+    Common::IPv4Header* ippart = (Common::IPv4Header*)&m_in_frame[14];
+    Common::TCPHeader* tcppart = (Common::TCPHeader*)&m_in_frame[0x22];
+
+    *hwpart = Common::EthernetHeader(IP_PROTOCOL);
+    hwpart->destination = network_ref[i].bba_mac;
+    hwpart->source = fake_mac;
+
+    *ippart = Common::IPv4Header(28, IPPROTO_TCP, network_ref[i].from, network_ref[i].to);
+
+    *tcppart = Common::TCPHeader(network_ref[i].from, network_ref[i].to, network_ref[i].seq_num,
+                                 network_ref[i].ack_num, 0x70 | TCP_FLAG_SIN | TCP_FLAG_ACK);
+    u8 options[] = {0x02, 0x04, 0x05, 0xb4, 0x01, 0x01, 0x01, 0x01};
+    memcpy(&m_in_frame[0x36], options, std::size(options));
+
+    // do checksum
+    tcppart->checksum = Common::ComputeTCPNetworkChecksum(network_ref[i].from, network_ref[i].to,
+                                                          tcppart, 28, IPPROTO_TCP);
+
+    network_ref[i].seq_num++;
+    target = sf::IpAddress(Common::swap32(ipdata->destination_addr));
```

Comment:
The declaration at the beginning of the function can be removed as it's only used here:
```C++
const sf::IpAddress target
```
I might be wrong about `const` here, so feel free to remove it if it complains about it.